internal/server/email/sendgrid.go

```diff
@@ -111,6 +106,18 @@ func SendTemplate(name, address string, template EmailTemplate, data any) error
 	}
 	msg.HTMLBody = w.Bytes()
 
+	if TestMode {
+		logging.Debugf("sent email to %q: %+v\n", address, data)
+		logging.Debugf("plain: %s", string(msg.PlainBody))
+		logging.Debugf("html: %s", string(msg.HTMLBody))
+		TestDataSent = append(TestDataSent, data)
+		return nil // quietly return
+	}
+
+	if len(SendgridAPIKey) == 0 {
+		return ErrNotConfigured
+	}
```

Comment:
seems like this is too far down in the function now

Reply:
I did this so that the email still gets constructed (i.e. you can tell what _would_ have been sent), otherwise it short circuits too quickly (and makes it harder to test).